src/AgentModel.cpp

```diff
@@ -655,35 +774,64 @@ void AgentModel::consciousFollow() {
     // get pointer to targets
     auto t = _input.targets;
 
-    // calculate net distance for following
-    unsigned long im = agent_model::NOT;
+    // calculate net distance for following targets
+    unsigned long im = agent_model::NOT;        // target on ego lane
+    unsigned long im_n_l = agent_model::NOT;    // target on neigbouring lane
+
+    int insert_idx;
     for (unsigned long i = 0; i < agent_model::NOT; ++i) {
 
         // ignore non-relevant targets (, ds < 0, other lane)
-        if (t[i].id == 0 || std::isinf(t[i].ds) || t[i].ds < 0.0 || t[i].lane != 0)
-            continue;
-
-        // check if distance is smaller
-        if (im == agent_model::NOT || t[im].ds > t[i].ds)
-            im = i;
+        if (t[i].id == 0 || std::isinf(t[i].ds) || t[i].ds < 0.0) continue;
+        
+        
+        if (t[i].lane == 0) {
 
+            // check if distance is smaller
+            if (im == agent_model::NOT || t[im].ds > t[i].ds)
+                im = i;
+        }
+        
+        // compute target of neigbouring lane only during lane changes
+        if (_lane_change_process_interval.getFactor() > 0 && t[i].lane == _state.decisions.laneChange) 
+        {
+            // check if distance is smaller
+            if (im_n_l == agent_model::NOT || t[im_n_l].ds > t[i].ds)
+                im_n_l = i;
+        }
     }
+    
+    // instantiate distance and velocity
+    double ds = INFINITY, v = 0.0, factor = 1;
 
-    // instantiate distance and  velocity
-    double ds = INFINITY, v = 0.0;
+    // closest ego lane target
+    factor = 1 - _lane_change_process_interval.getFactor();
     if (im != agent_model::NOT) {
         ds = t[im].ds - t[im].size.length * 0.5 - _param.vehicle.size.length * 0.5 + _param.vehicle.pos.x;
         v = t[im].v;
     }
-
+    
+    // save distance and velocity
+    _state.conscious.follow.targets[0].distance = ds;
+    _state.conscious.follow.targets[0].velocity = v;
+    _state.conscious.follow.targets[0].factor = factor;
+    
     // calculate if vehicle stands behind target vehicle
     bool standing = _input.vehicle.v < 1e-3 && v < 0.5 && ds <= _param.follow.dsStopped + 1e-2;
-
-    // save distance and velocity
-    _state.conscious.follow.distance = ds;
-    _state.conscious.follow.velocity = v;
     _state.conscious.follow.standing = standing;
+    
+    // closest neigbouring lane target
+    factor = _lane_change_process_interval.getFactor();
```

Comment:
Probably we should reset `ds` to infinity here? Since we use it in subconsciousFollow as break condition in the loop.

Reply:
Yes, but we also use the new introduced `factor` in subconsciousFollow. 

Nevertheless, I reset all values (ds, v, factor) from now on. Thanks for the input :-)